packages/action/src/run.ts

```diff
@@ -136,14 +136,31 @@ export async function run() {
 
   core.info('Got both sources');
 
-  const sources = {
-    old: new Source(oldFile, endpoint || `${schemaRef}:${schemaPath}`),
-    new: new Source(newFile, schemaPath),
-  };
+  let oldSchema: GraphQLSchema;
+  let newSchema: GraphQLSchema;
+  let sources: { new: Source; old: Source; }; 
+
+  if (extname(schemaPath.toLowerCase()) === ".json") {
+    oldSchema = buildClientSchema(JSON.parse(oldFile));
+    newSchema = buildClientSchema(JSON.parse(newFile));
```

Comment:
Introspection query results are normally part of `{ "data" : { ... } }` in the query output making this raise
```
Error: Invalid or incomplete introspection result. Ensure that you are passing "data" property of introspection response and no "errors" was returned alongside: { data: { __schema: [Object] } }.
```

Consider passing `JSON.parse(oldFile).data` to `buildClientSchema` if it exists in the parsed file

Reply:
@tvvignesh

Reply:
I will have a look at this today.

Reply:
@AllexVeldman @kamilkisiela Have raised a PR for the same here: https://github.com/kamilkisiela/graphql-inspector/pull/2059